feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go

```diff
@@ -0,0 +1,619 @@
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  true,
+			expectRead:   true,
+			expectStatus: 0,
+		}, {
+			desc:         "Secondary connection",
+			lowID:        *ygot.Uint64(90),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  false,
+			expectRead:   true,
+			expectStatus: 5,
+		}, {
+			desc:         "Primary reconnection",
+			lowID:        *ygot.Uint64(100),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  true,
+			expectRead:   true,
+			expectStatus: 0,
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			resp, err := streamP4RTArb(&test)
+			if err != nil && test.expectPass {
+				t.Errorf("Failed to setup P4RT Client: %v", err)
+			}
+			// Validate status code
+			if resp != test.expectStatus {
+				t.Errorf("Incorrect status code received: want %d, got %d", test.expectStatus, resp)
+			}
+			// Validate the response for Read/Write for each client
+			validateRWResp(t, &test)
+			// Disconnect Primary
+			removeClient(test.handle)
+		})
+	}
+}
+
+// Test Primary/Secondary clients
+func TestPrimarySecondary(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDeviceId(t, dut)
+	configurePortId(t, dut)
+	testCases := []testArgs{
+		{
+			desc:         "Primary connection",
+			lowID:        *ygot.Uint64(100),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  true,
+			expectRead:   true,
+			expectStatus: 0,
+		}, {
+			desc:         "Secondary connection",
+			lowID:        *ygot.Uint64(90),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  false,
+			expectRead:   true,
+			expectStatus: 6,
+		},
+	}
+	var p4Clients []*p4rt_client.P4RTClient
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			resp, err := streamP4RTArb(&test)
+			if err != nil && test.expectPass {
+				t.Errorf("Failed to setup P4RT Client: %v", err)
+			}
+			// Validate status code
+			if resp != test.expectStatus {
+				t.Errorf("Incorrect status code received: want %d, got %d", test.expectStatus, resp)
+			}
+			// Validate the response for Read/Write for each client
+			validateRWResp(t, &test)
+			// Keep track of connections for teardown
+			p4Clients = append(p4Clients, test.handle)
+		})
+	}
+	// Teardown clients
+	for _, p4Client := range p4Clients {
+		removeClient(p4Client)
+	}
+}
+
+// Test Duplicate ElectionID
+func TestDuplicateElectionID(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDeviceId(t, dut)
+	configurePortId(t, dut)
+	testCases := []testArgs{
+		{
+			desc:         "Primary connection",
```

Comment:
Can we only have the values that are changing in the table here and then add the constant items later in `for test:= range testCases{}` loop